csharp/tools/tracing-config.lua

```diff
@@ -48,6 +50,14 @@ function RegisterExtractorPack(id)
                     dotnetRunNeedsSeparator = true
                     dotnetRunInjectionIndex = i + 1
                 end
+                if arg == 'test' then
+                    testMatch = true
+                end
+                -- for `dotnet test`, we should not append `-p:UseSharedCompilation=false` to the command line
+                -- if a library or executable is being provided as an argument.
+                if arg:match('%.exe$') or arg:match('%.dll')  then
+                    libOrExe = true
+                end
```

Comment:
Alternatively
```suggestion
                if testMatch and (arg:match('%.exe$') or arg:match('%.dll'))  then
                    testMatch = false
                end
```
and then you can get rid of the `libOrExe` variable.

Reply:
I also considered that (and it can be simplified even further - we don't need to check that testMatch is true in the condition. However, I was a bit unsure on the effect on the logging code on l. 40, but I will change it based on your suggestion.